tools/fp20compiler/rc1.0_general.cpp

```diff
@@ -364,6 +375,13 @@ void OpStruct::Validate(int stage, int portion)
         if (RCP_ALPHA == portion &&
             RCP_RGB == reg[i].reg.bits.channel)
             errors.set("rgb register used in alpha portion");
+        if (i == 0 &&
+            RCP_RGB == portion &&
+            RCP_ALPHA == reg[i].reg.bits.channel)
+            errors.set("alpha register used as output in rgb portion");
+        if (i == 0 &&
+            RCP_BLUE == reg[i].reg.bits.channel)
+            errors.set("blue register used as output");

```

Comment:
Do these checks also work for implicit register specification?
Also missing checks to only allow BLUE instead of RGB in BLUETOALPHA.

**Expected behaviour:**

rgb portion:
```
spare0 = spare1.rgb; // allowed, same as spare0.rgb = spare1.rgb
spare0.a = spare1.rgb; // not allowed
spare0.a = spare1.b; // allowed for A/B and C/D only
spare0.b = spare1.rgb; // not allowed
spare0.rgb = spare1.rgb; // allowed
```
alpha portion:
```
spare0 = spare1.a; // allowed, same as spare0.a = spare1.a
spare0.a = spare1.a; // allowed
spare0.b = spare1.a; // not allowed
spare0.rgb = spare1.a; // not allowed
```